benchmarks/gluon_bench/main.go

```diff
@@ -22,6 +22,8 @@ var benches = []benchmarks.Benchmark{
 	benchmarks.NewMove(),
 	benchmarks.NewStore(),
 	benchmarks.NewExpunge(),
+	benchmarks.NewSearchText(),
```

Comment:
I think this will lead to a _lot_ of code duplication, since every combination of search keywords will need an additional benchmark type, with its own very similar setup/teardown... e.g. it's important to be able to benchmark complex search queries with multiple terms, and it'd be annoying having to create a new fixture for each combination of terms.

This isn't a perfect solution, but perhaps you could create a single search fixture type which accepts keywords to search over on construction? e.g. `NewSearch(SearchText(), SearchSince())`. And you could maybe even create aggregators, like `NewSearch(SearchOr(SearchText(), SearchSince()))`? Might be better.

Reply:
I have some refactoring planned up ahead to streamline implementation details such as this. If it's alright with you I would like to address this in the future.

Reply:
Sure, happy to wait :)